### ISSUE TITLE
coding details for a problem similar to DREAM challenge

### ISSUE THREAD

**Issue by TiaHao** [icc2115/Neural-GC#11]
Dear developer,

Thanks for sharing this amazing repository about the Neural Granger Causality project. I am dealing with a dataset containing 126 duplicates of small time series datasets representing data from 126 different cities, each of them having around 30 features and 100+ time steps. I am unsure how to apply cMLP or cLSTM for the whole dataset. I am thinking of separately training 126 models, but this would be too time-consuming. So I would appreciate it if there would be any suggestion or demo for how to implement the model for a challenge similar to DREAM datasets in the paper. Thank you very much!

Best,

**Comment by iancovert** [comment]
Thanks for checking out the package!

For this situation, would you prefer to train a single model with all of the 126 time series? If so, the training routines for each model (cMLP, cLSTM, etc) could treat them as independent time series, like different samples in a minibatch. You can see [here](https://github.com/iancovert/Neural-GC/blob/master/models/cmlp.py#L55) that the forward function has a leading batch dimension, and you can see in [this notebook](https://github.com/iancovert/Neural-GC/blob/master/clstm_lorenz_demo.ipynb) that our demos treat the data like a batch of one long time series.

Let me know how this sounds to you.

**Comment by TiaHao** [comment]
Thanks for your reply! If I understand correctly, in this way, a batch is one city's data right? One batch has dimension of (30, 100), 30: feature number, 100: time steps. in total, 126 batches.

**Comment by iancovert** [comment]
Yes, that's right. And it would mean that the temporal dynamics and connectivity between time series are shared across all cities. Does that sound like what you're looking for?

**Comment by TiaHao** [comment]
Yes! thanks a lot, that is what I am looking for! btw maybe one last question: would you know if I do not have GPU  acceleration ( I am using MacBook pro 2017, no M chip), how much time could it take to train the cMLP or cLSTM?

**Comment by iancovert** [comment]
I'm not exactly sure how long training will take, that depends on the dataset (how long each gradient step takes, how many steps are required to converge). cMLP is typically faster than cLSTM, so you might start with that. But training will definitely be slower on CPU than GPU, so it could be worth finding one, especially because you'll want to train with different lambda values to compare the results.

**Comment by TiaHao** [comment]
Thank you very much! I am going to try and test on my data!
